### PR TITLE
clean sogeBot update process

### PR DESCRIPTION
**Pull request by gOOvER**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### Changes to an existing Egg:

1. [x] Have you added an explanation of what your changes do and why you'd like us to include them?
2. [x] Have you tested your Egg changes?

### Changelog

- add database backup on reinstall
- add more folder cleanup to prevent errors on update
- fixed missing dotenv module

The egg need the PR https://github.com/parkervcp/yolks/pull/29, because buster have a too old sqllite version and throws missing sqlite errors.

### CODE REVIEW

File: bots/twitch/sogebot/egg-soge-bot.json [hunk 1]
```diff
@@ -17,13 +17,13 @@
     "config": {
         "files": "{\r\n    \".env\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"PORT\": \"{{server.build.default.port}}\"\r\n        }\r\n    }\r\n}",
         "startup": "{\r\n    \"done\": \"WebPanel is available at\"\r\n}",
-        "logs": "{}",
-        "stop": "^^C"
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"logs\/latest.log\"\r\n}",
+        "stop": "^C"
```
Comment (Software-Noob) [inline bots/twitch/sogebot/egg-soge-bot.json:21]: ^^C is a current workaround regarding catching termination signals when using eval for startup parsing. ^C doesn't get passed, and the server will be stuck stopping until Pterodactyl kills it

File: bots/twitch/sogebot/egg-soge-bot.json [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
...N}\" == \"latest\" ]; then\r\n    DOWNLOAD_LINK=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${MATCH})\r\nelse\r\n    VERSION_CHECK=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .tag_name')\r\n    if [ \"${VERSION}\" == \"${VERSION_CHECK}\" ]; then\r\n        DOWNLOAD_LINK=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .assets[].browser_download_url' | grep -i ${MATCH})\r\n    else\r\n        echo -e \"defaulting to latest release\"\r\n        DOWNLOAD_LINK=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url)\r\n    fi\r\nfi\r\n\r\necho -e \"Downloading from $DOWNLOAD_LINK\"\r\nwget -N $DOWNLOAD_LINK\r\nunzip -o $MATCH-*.zip\r\nrm -fR $MATCH-*.zip\r\n\r\necho \"--------------------------------------------------------------------------\"\r\necho \"Installing dependencies, this will take some time so sit back and relax...\"\r\necho \"IGNORE WARNINGS. THESE ARE NO ERRORS !!!\"\r\necho \"--------------------------------------------------------------------------\"\r\n\r\nnpm ci 2>&1 | tee npmlog.txt\r\n\r\necho -e \"install completed\"\r\nexit 0",
```
Comment (Software-Noob) [inline bots/twitch/sogebot/egg-soge-bot.json:25]: `no errors` should be `not errors`

**Review by Software-Noob — CHANGES_REQUESTED** [review]

File: bots/twitch/sogebot/egg-soge-bot.json [hunk 3]
```diff
@@ -17,13 +17,13 @@
     "config": {
         "files": "{\r\n    \".env\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"PORT\": \"{{server.build.default.port}}\"\r\n        }\r\n    }\r\n}",
         "startup": "{\r\n    \"done\": \"WebPanel is available at\"\r\n}",
-        "logs": "{}",
-        "stop": "^^C"
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"logs\/latest.log\"\r\n}",
```
[the remark this answers is not in the document]
Comment (gOOvER) [reply bots/twitch/sogebot/egg-soge-bot.json]: fixed

File: bots/twitch/sogebot/egg-soge-bot.json (hunk 2 above)
Comment (gOOvER) [reply bots/twitch/sogebot/egg-soge-bot.json]: fixed

**Review by Software-Noob — APPROVED** [review]
LGTM, thank you for contributing :+1:

**Pull request merged**